packages/neos-ui/src/Containers/Modals/NodeCreationDialog/index.js

```diff
@@ -50,14 +58,35 @@ export default class NodeCreationDialog extends PureComponent {
         secondaryInspectorComponent: null
     };
 
-    state = NodeCreationDialog.getDerivedStateFromProps(
-        this.props,
-        NodeCreationDialog.defaultState
-    );
+    constructor(props) {
+        super(props);
+
+        this.state = ((props, state) => {
```

Comment:
Todo: cleanup this code - we inlined getDerivedStateFromProps for testing - as this wouldnt work ...

Reply:
easiest fix would be extract it in a function outside of the class and use it here and also in  `static getDerivedStateFromProps`

Reply:
done via https://github.com/neos/neos-ui/pull/3200/commits/09fc4c41fb56a8756056b34010b8ba24e43e152f